Trace/TraceModel/TraceModel+Span.swift

```diff
@@ -301,6 +306,32 @@ extension TraceModel {
             
             return copy
         }
+        
+        // MARK: - Span validator
+        
+        /// :nodoc:
+        @discardableResult
+        public func spanValidator(span: Span) -> Bool {
```

Comment:
1. This needs to be an internal method as it is for internal use only
2. I don't understand why it requires a span object when it's an instance method. wouldn't `func validate() -> Bool` do the same. 
3. if `.end` is not present shouldn't that return false as it isn't in the correct state.

Reply:
Thanks all valid and makes sense to me. Updated.